internals/cli/masker/writer.go

```diff
@@ -122,7 +122,16 @@ func (mw *MaskedWriter) Write(p []byte) (n int, err error) {
 	return len(p), nil
 }
 
-func (mw *MaskedWriter) write() {
+// process receives incoming bytes from calls to Write (over the incoming bytes channel) masks them if necessary and passes
+// them through to Run (over the output channel).
+//
+// - When the incoming bytes do not end in a partial secret, they are directly passed to the output channel.
+// - When the incoming bytes do end in a partial secret, all bytes up to the partial secret are passed to the
```

Comment:
What happens when the incoming bytes **start** with a partial secret? Your phrasing here suggests that is a different scenario while I feel they may actually be the same: when the incoming bytes match a partial secret?

Reply:
When incoming bytes start with a partial secret and other bytes that do not finish the secret come after, we know that the partial secret is not the secret, so we will output the plaintext of the partial secret.

For example:
There's one secret with the value `foo`

When the value `fbar` is written, all bytes are passed though (and outputted by the `Run` process), including the partial secret `f`.
When the value `barf` is written, `bar` is written, but the partial secret `f` is not yet. It is processed as described in the comment.

Does that clarify things? Do you feel the comment should be adapted to make this clearer?